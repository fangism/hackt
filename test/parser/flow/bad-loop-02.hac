// "bad-loop-03.hac"
// expect graceful syntax error failure
// reduced test case, originally reported by Clint Kelly
// missing range expression of loop

(;i: foo[0](); )

