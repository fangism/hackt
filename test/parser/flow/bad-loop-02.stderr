Parse error: base reference of function call must be an id_expr, but got: (index-expr) at [6:6..10]
At: "bad-loop-02.hac":6:
parse error: syntax error
parser stacks:
state	value
#STATE#	(null) 
#STATE#	( [6:1]
#STATE#	; [6:2]
#STATE#	identifier: i [6:3]
#STATE#	: [6:4]
#STATE#	(chp-func-call) [6:12..13]
#STATE#	; [6:14]
in state #STATE#, possible rules are:
	range : expr . RANGE expr  (#RULE#)
	range : expr .  (#RULE#)
acceptable tokens are: 
	':' (reduce)
	']' (reduce)
	RANGE (shift)
